src/app/auth/components/singup/singup.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { SignUpBody } from 'src/app/core/types/auth.types';
+import { AuthService } from '../../services/auth.service';
+@Component({
+  selector: 'app-singup',
+  templateUrl: './singup.component.html',
+  styleUrls: ['./singup.component.scss'],
+})
+export class SingupComponent {
+  patterns = {
+    MIN_LENGTH: 4,
+    MAX_LENGTH: 20,
+    PATTERN_NAME: /^[a-z0-9]+$/,
+    PATTERN_PASSWORD:
+      /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
```

Comment:
The same password pattern as above. So I would recommend move it to shared constant ( shared/constants/patterns.constant.ts)